app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java

```diff
@@ -73,21 +75,40 @@ private void releaseWakeLock() {
             wakeLock.release();
         } catch (Exception e) {/* ignore */}
         wakeLock = null;
+        wakeLockAcquireTimestampMs = 0;
+        wakeLockTimeoutMs = 0;
     }
 
-    private void acquireWakeLock() {
+    public long getWakeLockTimeout() {
+        return (wakeLock == null || !wakeLock.isHeld() || wakeLockAcquireTimestampMs <= 0 || wakeLockTimeoutMs <= 0)
+                ? 0
+                : wakeLockAcquireTimestampMs + wakeLockTimeoutMs - SystemClock.elapsedRealtime();
+    }
+
+    public void acquireWakeLock(long msTimeout) {
         releaseWakeLock();
 
+        if (msTimeout <= 0) {
+            return;
```

Comment:
How about `Logger.warn` here or `.info` before this control flow to leave a message what number the user tried to set? It helps to track the configurable number.

Reply:
makes sense, will add